modules/siddhi-core/src/main/java/org/wso2/siddhi/core/SiddhiAppRuntime.java

```diff
@@ -693,4 +693,24 @@ public void enableStats(boolean statsEnabled) {
             log.debug("Siddhi App '" + getName() + "' statistics reporting not changed!");
         }
     }
+
+    /**
+     * To enable and disable Siddhi App playback mode on runtime.
+     *
+     * @param playBackEnabled whether statistics is enabled or not
+     */
+    public void enablePlayBack(boolean playBackEnabled) {
+        this.siddhiAppContext.setPlayback(playBackEnabled);
```

Comment:
When moving from play back enabled to disabled need to start the schedulers and vice versa.

Reply:
When siddhi app runtime gets created , we create the Scheduler class where in the constructor we initialize the listener to listen events coming from stream junction (which they only will be called if isPlayBack is true).  If isPlayBack is false, the schedule method of Scheduler class will create a scheduledExecutorService to inform in the corresponding time which will run once. 

So don't we only need to just change the the boolean value. According to the boolean schedular's will work. We do not need to stop the scheduledExecutorService in schedule method because schedule will not call if isPlayBack is true. 

Are you suggesting we need to stop the last scheduled job if we change to isPlayBack to true ? (meaning we now want the system to work on event times. not scheduled time) A side question then if we are to stop the scheduledExecutorService then we should not get the schedular from siddhi app context. we will have to create another scheduler .

But for this scenario, (passive becomes active) we do not have method schedule called yet thus 
scheduledExecutorService is not created. So when we first make isPlayBack to true. It will get the events from stream junction. And once done sending , when we switch back to isPlayBack to false, it will create the scheduledExecutorService to work with current time. So we do not have a stop scenario in this case.